src/raw_node.rs

```diff
@@ -196,6 +192,63 @@ impl Ready {
     }
 }
 
+/// ReadyRecord encapsulates some needed data from the corresponding Ready.
+#[derive(Default, Debug, PartialEq)]
+struct ReadyRecord {
+    number: u64,
+    // (index, term) of the last entry from the entries in Ready
+    last_entry: Option<(u64, u64)>,
+    snapshot: Snapshot,
+    messages: Vec<Message>,
+}
+
+/// PersistLastReadyResult encapsulates the commit index, committed entries and
+/// messages that are ready to be applied or be sent to other peers.
+#[derive(Default, Debug, PartialEq)]
+pub struct PersistLastReadyResult {
+    commit_index: Option<u64>,
```

Comment:
Why it could be None?

Reply:
You can see the `fn commit_index` annotation above.